pkg/ebpf/bpf_client.go

```diff
@@ -583,49 +673,66 @@ func (l *bpfClient) AttacheBPFProbes(pod types.NamespacedName, podIdentifier str
-	podNamespacedName := utils.GetPodNamespacedName(pod.Name, pod.Namespace)
+	numInterfaces = actualInterfaceCount
 
-	if !isIngressProbeAttached {
-		progFD, err := l.attachIngressBPFProbe(hostVethName, podIdentifier)
-		duration := msSince(start)
-		sdkAPILatency.WithLabelValues("attachIngressBPFProbe", fmt.Sprint(err != nil)).Observe(duration)
+	for index := 0; index < numInterfaces; index++ {
+		// We attach the TC probes to the hostVeth interfaces of the pod. Derive the hostVeth name from the Name and Namespace of the Pod.
+		// Note: The below naming convention is tied to VPC CNI and isn't meant to be generic
+		hostVethName, err := utils.GetHostVethName(pod.Name, pod.Namespace, index, []string{POD_VETH_PREFIX, BRANCH_ENI_VETH_PREFIX})
 		if err != nil {
-			log().Errorf("Failed to Attach Ingress TC probe for pod: %s in namespace %s error: %v", pod.Name, pod.Namespace, err)
-			sdkAPIErr.WithLabelValues("attachIngressBPFProbe").Inc()
+			log().Warnf("Failed to attach ebpf probes for pod %s in namespace %s. Pod might have been deleted", pod.Name, pod.Namespace)
 			return err
 		}
-		log().Infof("Successfully attached Ingress TC probe for pod: %s in namespace %s", pod.Name, pod.Namespace)
-		l.IngressPodToProgMap.Store(podNamespacedName, progFD)
-		currentPodSet, _ := l.IngressProgToPodsMap.LoadOrStore(progFD, make(map[string]struct{}))
+
+		log().Infof("AttacheBPFProbes for pod %s in namespace %s with hostVethName %s at interface %d", pod.Name, pod.Namespace, hostVethName, index)
+
+		if !isIngressProbeAttached {
+			start := time.Now()
+			ingressProgFD, err = l.attachIngressBPFProbe(hostVethName, podIdentifier)
+			duration := msSince(start)
+			sdkAPILatency.WithLabelValues("attachIngressBPFProbe", fmt.Sprint(err != nil)).Observe(duration)
+			if err != nil {
+				log().Errorf("Failed to Attach Ingress TC probe for pod: %s in namespace %s at interface %d error: %v", pod.Name, pod.Namespace, index, err)
+				sdkAPIErr.WithLabelValues("attachIngressBPFProbe").Inc()
+				return err
+			}
+			log().Infof("Successfully attached Ingress TC probe for pod: %s in namespace %s at interface %d", pod.Name, pod.Namespace, index)
+		}
+
+		if !isEgressProbeAttached {
+			start := time.Now()
+			egressProgFD, err = l.attachEgressBPFProbe(hostVethName, podIdentifier)
+			duration := msSince(start)
+			sdkAPILatency.WithLabelValues("attachEgressBPFProbe", fmt.Sprint(err != nil)).Observe(duration)
+			if err != nil {
+				log().Errorf("Failed to Attach Egress TC probe for pod: %s in namespace %s at interface %d error: %v", pod.Name, pod.Namespace, index, err)
+				sdkAPIErr.WithLabelValues("attachEgressBPFProbe").Inc()
+				return err
+			}
+			log().Infof("Successfully attached Egress TC probe for pod: %s in namespace %s at interface %d", pod.Name, pod.Namespace, index)
+		}
+	}
+	if !isIngressProbeAttached {
+		l.IngressPodToProgMap.Store(podNamespacedName, ingressProgFD)
```

Comment:
usage of `ingressProgFd` and `egressProgFd` is a little unclear. They are declared outside the `for` loop, thus the FD value will be corresponding to last interface attached. 

I feel there is some blurring of boundary of multiple FDs and we are storing `1` of those to deduce conditions like attachedProbeToPod. 

I am not following recovery path (during NPA re-start) and deletion path on how we are cleaning up probes attached to other NICs of the pod. Can you confirm that once? 

to make sure we add probes to all interfaces post NPA re-start and clean them all during deletion flow.

Reply:
Update - `ProgFD` and `ProgID` are PodIdentifier level properties (Not per Pod or interface per pod).